### PR TITLE
feat: ignore accept connection request (AR-1462)

### PR DESCRIPTION
**Pull request by Garzas**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1462" title="AR-1462" target="_blank">StoryAR-1462</a>  Ignore connection request
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

_Briefly describe the issue you have solved or implemented with this pull request. If the PR contains multiple issues, use a bullet list._

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-639/1/) **failed**.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-639/2/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreen.kt
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreen.kt:55]: Maybe, if possible and not too time demanding, we could change this to method reference, since @trOnk12 discovered a nice performance improvement, between calling lambdas vs method references

`ignoreConnectionRequest = viewModel::acceptConnectionRequest`

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationRouter.kt
Comment (yamilmedina) [reply app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationRouter.kt:58]: I don't see a problem with observing this other state, but, maybe @trOnk12  can take a look, since he recently refactored some of this code 👨‍🔬  

I can imagine that passing directly `conversationState.conversationSheetContent` wasn't an option ?

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/MentionScreen.kt
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/home/conversationslist/MentionScreen.kt:26]: maybe to follow the same pattern, we can rename it to 
```suggestion
    onOpenConversationNotificationsSettings: (ConversationItem) -> Unit,
```

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/CallsScreen.kt
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/home/conversationslist/CallsScreen.kt:25]: maybe to follow the same pattern, we can rename it to
```suggestion
    onOpenConversationNotificationsSettings: (ConversationItem) -> Unit,
```

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/AllConversationScreen.kt
Comment (yamilmedina) [inline app/src/main/kotlin/com/wire/android/ui/home/conversationslist/AllConversationScreen.kt:27]: maybe to follow the same pattern, we can rename it to
```suggestion
    onOpenConversationNotificationsSettings: (ConversationItem) -> Unit,
```

**Review by yamilmedina — COMMENTED** [review]
I left some comments, I think when we sorted those doubts in a brief call, so we can have a look. Good work as always, I tested this locally and worked perfectly 🚀  👨‍🍳

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-639/3/) **failed**.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-639/4/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreen.kt
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/ui/userprofile/other/OtherUserProfileScreen.kt:55]: done

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/MentionScreen.kt
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/ui/home/conversationslist/MentionScreen.kt:26]: done

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/CallsScreen.kt
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/ui/home/conversationslist/CallsScreen.kt:25]: done

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/AllConversationScreen.kt
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/ui/home/conversationslist/AllConversationScreen.kt:27]: done

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-639/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3528571)(3528571)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3528571)(3528571)-dev-release.apk)

File: app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationRouter.kt
Comment (trOnk12) [reply app/src/main/kotlin/com/wire/android/ui/home/conversationslist/ConversationRouter.kt:58]: I also do not see a problem I think, it gives us a option to control the state of it outside the parent so I think it is even better

**Review by trOnk12 — APPROVED** [review]
looking good to me ! :) but since @yamilmedina was also looking into it, maybe lets wait for his approval as well ?

**Review by yamilmedina — APPROVED** [review]
greeeat! thanks for the changes, good work as always 🙌

**Comment by AndroidBob** [comment]
Build [7](https://10.10.124.134/job/wire-android-reloaded/job/PR-639/7/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(3528754)(3528754)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(3528754)(3528754)-dev-release.apk)

**Pull request merged**
